packages/webpack-plugin/src/federatedStatsCollector.js

```diff
@@ -0,0 +1,75 @@
+
+const extractSharedModuleOpts = (moduleIdentifier) => {
+  const [
+    _issuerType, _shareScope, _shareKey, semver, _strictVersion, _fullPath, singleton, eager
+  ] = moduleIdentifier.split(/(?<!\|)\|(?!\|)/);
+
+  return {
+    semver: semver === 'undefined' ? '*' : semver,
+    singleton: singleton === 'true',
+    eager: eager === 'true',
+  };
+}
+
+const isSharedModule = (moduleIdentifier) => moduleIdentifier.indexOf(`webpack/sharing/`) === 0
+
+const collectSharedModuleConsumptionStats = (modules = []) => {
+  const sharedConsumes = {};
+  const sharedModules = {};
+
+  modules.forEach((module) => {
```

Comment:
dark magic. I have absolutely no idea what's happening here.

Reply:
Heh, worth writing a huge comment or a link to github explaination. Thats what took me most of the time, analyzing stats to come up with a way to extract module shared dependencies. They happen to be stored under module.reasons